### DISCUSSION TITLE
Runtime: Unsupported Hardware Type

### DISCUSSION THREAD

**Discussion by MichuPicchu** [dashingsoft/pyarmor#1514] [Q&A]
Hi, just before I activate my group license, I am checking whether my device is suitable for pyarmor. I am using a Lenovo ThinkEdge device.

```
$ pyarmor gen -O dist --assert-call foo.py
INFO     Python 3.10.13
INFO     Pyarmor 8.3.10 (trial), 000000, non-profits
INFO     Platform linux.aarch64
```

I am able to run my obfuscation script within host and in a docker container.

However, when I run the following on host,

```
$python3 -c"from pyarmor.cli.core.pytransform3 import get_hd_info; print(get_hd_info(21))"
```

I get the following error:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
RuntimeError: Unsupported hardware type
```

However, I can run this exact command from within my docker, and seem to be able to get the same output even if I restarted the docker container.

How can I solve this runtime error on my host?

Any help would be greatly appreciated.
Thanks

**Comment by MichuPicchu** [comment]
Not sure how helpful this is, but when I run 

**pyarmor-7 hdinfo** on my host, it fails to get harddisk information and also fails to get ip address.
However, this command also doesn't work in docker, but am still able to run print(get_hd_info(21)).

**Comment by jondy** [comment]
First this error message `Unsupported hardware type` only reported by old Pyarmor, new version should never report this error message.
So test your python3 version by `python3 --version`
Uninstall pyarmor totally by `pip uninstall pyarmor pyarmor.cli.core` and `rm -rf /path/to/package/pyarmor/`
Install pyarmor again, and try to run `python3 -c"xxxx"` again to test
Also test `pyarmor reg -g 1`
If failed, please paste pip install log and pyarmor output log also.
And try to test these by root

**Comment by MichuPicchu** [comment]
Thanks a lot. Seems to work now after reinstalling pyarmor.
